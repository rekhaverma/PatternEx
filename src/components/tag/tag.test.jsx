import React from 'react';
import Tag from './tag.component';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<Tag />', () => {
  let props;

  beforeEach(() => {
    props = {
      label: 'Unit Testing'
    };
  });

  it('should match with snapshot - default props', () => {
    const wrapper = shallow(<Tag {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match with snapshot - with remove button', () => {
    props.onRemove = jest.fn();
    const wrapper = shallow(<Tag {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onRemove when element is clicked', () => {
    props.onRemove = jest.fn();
    const wrapper = shallow(<Tag {...props} />);
    wrapper.find('.icon-Trash-icon').simulate('click');
    expect(props.onRemove.mock.calls.length).toBe(1);
  });
});
