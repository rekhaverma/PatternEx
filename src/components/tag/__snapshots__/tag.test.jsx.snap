// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should match with snapshot - default props 1`] = `
<div
  className="advanced-tag"
>
  <span>
    Unit Testing
  </span>
</div>
`;

exports[`<Tag /> should match with snapshot - with remove button 1`] = `
<div
  className="advanced-tag"
>
  <span>
    Unit Testing
  </span>
  <span
    className="advanced-tag__remove icon-Trash-icon"
    onClick={[Function]}
  />
</div>
`;
